src/Connector.sol

```diff
@@ -7,11 +7,15 @@ import { RestrictedTokenLike } from "./token/restricted.sol";
 import { MemberlistLike } from "./token/memberlist.sol";
 
 interface RouterLike {
-    function sendMessage(uint32 destinationDomain, uint64 poolId, bytes16 trancheId, uint256 amount, address user) external;
+    function sendMessage(uint64 poolId, bytes16 trancheId, uint256 amount, address user) external;
 }
 
 contract CentrifugeConnector {
 
+    enum Domains {
```

Comment:
Not sure I understand. Your enums resolve to `0x000100000000000000`, for example?

Reply:
> Not sure I understand. Your enums resolve to 0x000100000000000000, for example?


Each Domain variant, i.e, `Domain::EVM(_)` or `Domain::Parachain(_)` will encode different. See the individual tests linked above for each variant and with different values. The `Moonbeam` domain encodes to `0x0100`, the ethereum mainnet encodes to `0x000100000000000000`, etc. On the cent-chain we currently don't handle `Centrifuge` but we should add it in order to be able to decode `Transfer` messages with `domain` set `Centrifuge` (again, if we keep that field in that message)

Reply:
@offerijns How could this be implemented in Solidity? I don't think we have support for enum variants

Reply:
I don't know, that needs some research! Simplest solution would be manually encoding it the same way as cnet chain. So `bytes(uint8(0) )+ (uint64(evmChainId))` for an EVM chain id basically and `bytes(uint8(1) )+ (uint64(centchainIarachainId))` for Cent Chain.

Reply:
Then Alina's `DomainLookup` that we removed might be the best approach to solve this. Our reasoning for removing it was that we didn't want the extra SLOAD, but I think in order to use an enum pattern that resolves to real data (instead of just integers), it will always involve an SLOAD.

Reply:
I think the function interface can just be
```solidity
enum Domain {EVM, Parachain}

function transfer(
    uint64 poolId,
    bytes16 trancheId,
    address user,
    uint256 amount,
    Domain destinationDomain,
    uint destinationDomainId
) public {}
```

Reply:
Alright just pushed an update for this I think we'll all be happy with. Still need to work on the encoding though. @NunoAlexandre maybe we can work on this together tomorrow. I'm not really sure how you're encoding the chainIDs to match whats show in your tests.

Reply:
@AStox sounds good! For the example test you wrote for Avalanche, I see that the Avalanche chain Id encodes as little endian on Solidity, so we need to do that on the cent-chain as well (we had to do that for a couple of other uint values already). Notice how on the rust side it encodes to `6aa80` while on the solidity side it is `a86a` (reverse byte order)